src/checker.c

```diff
@@ -474,6 +474,10 @@ Conversion conversion(Type *type, Type *target) {
         return result;
     }
 
+    if (type->kind == TypeKind_Enum && IsInteger(target)) {
+        return ConversionKind_None;
```

Comment:
I think we should have a `ConversionKind` for this incase some backend want's to implement Enums in some special way